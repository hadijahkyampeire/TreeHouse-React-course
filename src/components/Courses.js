import React from 'react';
import {
  BrowserRouter, Route, Redirect
} from 'react-router-dom';

import CourseContainer from './courses/courseContainer';
import { HTMLCourses, CSSCourses, JSCourses } from '../data/courses';

const Courses = ({match}) => (
  <div className="main-content courses">
    <div className="course-header group">
      <h2>Courses</h2> 
      <ul className="course-nav">
        <li><a href={`${match.url}/html`}>HTML</a></li>
        <li><a href={`${match.url}/css`}>CSS</a></li>
        <li><a href={`${match.url}/javascript`}>JavaScript</a></li>
      </ul>
    </div>
    <BrowserRouter>
    <div>
   
    <Route exact path={match.path} render={()=>  <Redirect to={`${match.path}/html`}/>}/>
    <Route path={`${match.path}/html`} render={() => <CourseContainer data={HTMLCourses}/>}/>
    <Route path={`${match.path}/css`} render={() => <CourseContainer data={CSSCourses}/>}/>
    <Route path={`${match.path}/javascript`} render={() => <CourseContainer data={JSCourses}/>}/>
    </div>
    </BrowserRouter>
    {/* Write routes here... */}

  </div>
);

export default Courses;