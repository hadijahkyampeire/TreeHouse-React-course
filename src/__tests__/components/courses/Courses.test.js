import React from 'react';
import enzyme from 'enzyme';
import Courses from '../../../components/courses/Course';

describe('About component', () => {
    const node = enzyme.shallow(<Courses/>);

    it('should render without crashing', () => {
        expect(node).toMatchSnapshot();
    });
});