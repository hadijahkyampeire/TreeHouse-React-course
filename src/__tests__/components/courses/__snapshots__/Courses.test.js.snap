// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Course />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="course"
          className="course-img"
          src={undefined}
/>,
        <div>
          <h3 />
          <p />
</div>,
      ],
      "className": "course media group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "course",
          "className": "course-img",
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3 />,
            <p />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="course"
            className="course-img"
            src={undefined}
/>,
          <div>
            <h3 />
            <p />
</div>,
        ],
        "className": "course media group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "course",
            "className": "course-img",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3 />,
              <p />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
